### ISSUE TITLE
[QA] integrity checker complains about config.json

### ISSUE THREAD

**Issue by jnweiger** [owncloud/web#4530]
Deployed web-1.0.0 into docker run -ti --rm -v $HOME/oc10-mnt-data:/mnt/data -p 8080:8080 owncloud/server:10.6.0
using https://owncloud.github.io/clients/web/deployments/oc10-app/

According to https://owncloud.github.io/clients/web/deployments/oc10-app/
it is necessary to edit the file web/config.json

This file is covered by the integrity checker, and leads to complaints after editing. BAD

Expected behaviour: 
* config.json should be exempt from integrity checking. 
* config.json should be preserved between updates. (not sure if it is?)

```
Array
(
    [web] => Array
        (
            [INVALID_HASH] => Array
                (
                    [config.json] => Array
                        (
                            [expected] => 38cf76fd0e2e3be11d3e4c8c4651619551078474a4458bbf71d3eee9e6bf8b1eb6d54913b3de26a29d0f6065dc101f8b46b0f191f6ef2626e3ba111b71a3e0e0
                            [current] => f5927fed0e569003ddbb959017510b9a283c2567cb66211f998f0b7cf6616a8a0b48118667b3071a0585c2061a8f9d2d3ac104a934f20461ea9474e59cc5fd1a
                        )
```

**Comment by micbar** [comment]
yes. Expected. I also think we need to change it.

@kulmann @LukasHirt can we move the config.json out of the `web` folder?

**Comment by LukasHirt** [comment]
> can we move the config.json out of the web folder?

The location is currently hardcoded. Would need to think if it'd be possible to have some dynamic location.

**Comment by kulmann** [comment]
Do we have a chance to find out the oc10 url? Because then we could add a php controller that serves the content of the config.json file from within the config folder of oc10.

**Comment by LukasHirt** [comment]
> Do we have a chance to find out the oc10 url?

We could use the base URL that's visited, couldn't we? @kulmann

**Comment by kulmann** [comment]
Hmm.. `window.location.origin`

**Comment by kulmann** [comment]
Ok, I'll try it out. Basic plan: when the `config.json` is not there, we try to call the php controller for loading the content of the config.json.

**Comment by spupuz** [comment]
in my case integrity checker also show missing .htaccess and config.json bad has, can't see new web interface i do also followed the guide.

**Comment by micbar** [comment]
can you post the details @spupuz 
We can help you

Some other people stumbled over this too.

We need to improve the guide ;-)

**Comment by jnweiger** [comment]
Confirmed fixed in 1.0.2
* no complains from interirty checker
* the guide has
![image](https://user-images.githubusercontent.com/1108546/105258078-3ef7b380-5b89-11eb-9173-64a1b9cb1fc7.png)
